### PR TITLE
IRSDemoTest: wait until transactions have finished before completing …

### PR DESCRIPTION
**Pull request by adagys**
…the test.

The date to set the demo date to is now + 6 months.

### CODE REVIEW

File: samples/irs-demo/src/integration-test/kotlin/net/corda/irs/IRSDemoTest.kt [hunk 1]
```diff
@@ -5,45 +5,75 @@ import com.typesafe.config.Config
...
+class IRSDemoTest : IntegrationTestCategory {
     fun Config.getHostAndPort(name: String): HostAndPort = HostAndPort.fromString(getString(name))!!
+    val rpcUser = User("user", "password", emptySet())
+    val currentDate = LocalDate.now()
+    val futureDate = currentDate.plusMonths(6)
 
     @Test fun `runs IRS demo`() {
-        driver(dsl = {
+        driver(useTestClock = true, isDebug = true) {
             val controller = startNode("Notary", setOf(ServiceInfo(SimpleNotaryService.type), ServiceInfo(NodeInterestRates.type))).getOrThrow()
```
Comment (shamsasari) [inline samples/irs-demo/src/integration-test/kotlin/net/corda/irs/IRSDemoTest.kt]: If you use `Futures.allAsList` you can start all three nodes in parallel - speed things up a bit.

File: samples/irs-demo/src/main/kotlin/net/corda/irs/flows/UpdateBusinessDayFlow.kt [hunk 2]
```diff
@@ -51,7 +51,7 @@ object UpdateBusinessDayFlow {
         @Suspendable
         override fun call(): Unit {
             progressTracker.currentStep = NOTIFYING
-            for (recipient in serviceHub.networkMapCache.partyNodes) {
+            for (recipient in serviceHub.networkMapCache.partyNodes.sortedBy { it.legalIdentity.name }) {
```
Comment (adagys) [inline samples/irs-demo/src/main/kotlin/net/corda/irs/flows/UpdateBusinessDayFlow.kt]: actually this doesn't fix the timestamp invalid problem, I was just able to reproduce it. On the plus side (?) the test now hangs if that happens..

File: samples/irs-demo/src/integration-test/kotlin/net/corda/irs/IRSDemoTest.kt [hunk 3]
```diff
@@ -5,45 +5,75 @@ import com.typesafe.config.Config
...
+            val nodeA = startNode("Bank A", rpcUsers = listOf(rpcUser)).getOrThrow()
             val nodeB = startNode("Bank B").getOrThrow()
+            val nextFixingDates = getFixingDateObservable(nodeA.config)
+
             runUploadRates(controller.config.getHostAndPort("webAddress"))
             runTrade(nodeA.config.getHostAndPort("webAddress"))
+            // Wait until the initial trade and all scheduled fixings up to the current date have finished
+            nextFixingDates.first { it == null || it > currentDate }
+
             runDateChange(nodeB.config.getHostAndPort("webAddress"))
```
Comment (adagys) [inline samples/irs-demo/src/integration-test/kotlin/net/corda/irs/IRSDemoTest.kt]: Actually I doesn't really matter who we send it to, they run the same protocol & send the new date in the same order

File: samples/irs-demo/src/integration-test/kotlin/net/corda/irs/IRSDemoTest.kt (hunk 1 above)
Comment (adagys) [inline samples/irs-demo/src/integration-test/kotlin/net/corda/irs/IRSDemoTest.kt]: we use this pattern all over the place, I can add a TODO to look into driver node start parallelisation in general

**Comment by adagys** [comment]
Recipient ordering seems to fix the invalid timestamp issue

**Comment by mikehearn** [comment]
Needs a rebase. @adagys ball in your court.

**Pull request merged**